test/kitchen/docs/README.md

```diff
@@ -1,42 +1,128 @@
+- scenarios: `kitchen_scenario_<OS>_aX` which is a combination of a chosen OS and major Agent version.
+
+**Note:** To avoid having too many different test types, we bind each test suite with a specific Azure location. Eg. all install script tests are run in the same Azure location.
+
+The real jobs that are run in the Gitlab pipelines are thus a combination of test types and scenarios, named `kitchen_<OS>_<test_suite>_<flavor>-aX`.
+
+##### testkitchen_cleanup
+
+The `testkitchen_cleanup_azure-aX` jobs clean up the Azure VMs that were used during the `testkitchen_testing` stage.
+
+**Note:** The kitchen test jobs send a `destroy` request to Azure when their tests end, so usually these jobs do nothing. They're mainly useful to make sure all VMs have been destroyed, or when kitchen jobs are cancelled while they're running.
+
+**Note:** Azure VMs are also cleaned up on a daily basis by a job in the `mars-jenkins-scripts` pipeline.
+
+The `testkitchen_cleanup_s3-aX` jobs clean up the S3 buckets of the testing repos.
+
+**Important note:** These jobs are always run, even when a kitchen test fails. This is done on purpose (to avoid having remaining resources once a job fails).
+The downside of this approach is that when you want to retry a failing ktichen test job, you'll have to re-run the corresponding deploy job first (to put the Agent packages back in the testing repository).
+
+#### Test suites
+
+kitchen tests are defined in a file named `kitchen.yml` which defines:
+- the provisioner (`provisioner:` key) used in the tests. In the pipeline, `chef-solo` is used.
+- the driver (`driver:` + `driver_config:` keys) used to run the tests. In the pipeline, the [Azure kitchen driver](https://github.com/test-kitchen/kitchen-azurerm) is used.
+
+- the list of platforms (`platforms:` key) used in the tests.
+- the list of test suites (`suites:` key) to run.
+
+To reduce duplication, the `kitchen.yml` file is generated at test time by the `run-test-kitchen.sh` script ([see below](#test-execution-tasks/run-test-kitchen.sh)) by concatenating 2 template files:
+
+- the `kitchen-azure-common.yml` file, which defines the provisioner, driver, and platforms that will be used. It also defines a number of useful variables that can be used by the test suite templates (such as: which major Agent version is used, what are the testing repository URLs, etc.).
+- a `kitchen-azure-<test_suite>-test.yml` file, which contains the test suite definition for a particular test. That definition consists in the list of recipes to run, with their attributes, and optionally a list of platforms to not run the test on (eg. if you know that a particular test x platform combination cannot work).
+
+These files are `.erb` template files, the rendered contents (eg. the list of platforms) depend on what variables are passed to the `run-test-kitchen.sh` script.
+
+#### Test execution (tasks/run-test-kitchen.sh)
```

Comment:
I think it could be clearer if each of the environment variables here were in a list item/description item/its own subsection, with the current style at first I thought the only environment variable was `TEST_PLATFORMS` and I got confused.